src/main/java/uk/nhs/digital/nhsconnect/nhais/service/edifact_to_fhir/DrugsMarkerExtensionPatchTransactionMapper.java

```diff
@@ -0,0 +1,42 @@
+package uk.nhs.digital.nhsconnect.nhais.service.edifact_to_fhir;
+
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.Transaction;
+import uk.nhs.digital.nhsconnect.nhais.model.fhir.DrugsMarkerExtension;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentBooleanExtension;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentPatch;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentPatchOperation;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentValue;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class DrugsMarkerExtensionPatchTransactionMapper implements PatchTransactionMapper {
+
+    @Override
+    public AmendmentPatch map(Transaction transaction) {
+        var drugMarker = transaction.getDrugsMarker();
+        if (drugMarker.isPresent()) {
+            var drugMarketBoolean = drugMarker.get().getDrugsMarker();
+            return createAmmendmentExtensionPatch(drugMarketBoolean);
+        } else {
+            return null;
+        }
+    }
+
+    private AmendmentPatch createAmmendmentExtensionPatch(boolean element) {
+        var path = "/extension/0";
+
+        AmendmentPatch amendmentPatch = new AmendmentPatch();
+        if (!element){
+            amendmentPatch.setOp(AmendmentPatchOperation.REMOVE);
+            amendmentPatch.setPath(path);
+            amendmentPatch.setValue(new AmendmentBooleanExtension(DrugsMarkerExtension.URL, "false"));
+        } else {
+            amendmentPatch.setOp(AmendmentPatchOperation.REPLACE);
+            amendmentPatch.setPath(path);
+            amendmentPatch.setValue(new AmendmentBooleanExtension(DrugsMarkerExtension.URL, "true"));
+        }
+
+        return amendmentPatch;
+    }
```

Comment:
````java
@Override
public AmendmentPatch map(Transaction transaction) {
    return transaction.getDrugsMarker()
        .map(DrugsMarker::getDrugsMarker)
        .map(this::createAmendmentExtensionPatch)
        .orElse(null);
}

private AmendmentPatch createAmendmentExtensionPatch(boolean drugMarker) {
    if (drugMarker){
        return createReplaceOperation();
    }
    return createRemoveOperation();
}

private AmendmentPatch createReplaceOperation() {
    var path = "/extension/0";
    AmendmentPatch amendmentPatch = new AmendmentPatch();
    amendmentPatch.setOp(AmendmentPatchOperation.REPLACE);
    amendmentPatch.setPath(path);
    amendmentPatch.setValue(new AmendmentBooleanExtension(DrugsMarkerExtension.URL, "true"));
    return amendmentPatch;
}

private AmendmentPatch createRemoveOperation() {
    var path = "/extension/0";
    AmendmentPatch amendmentPatch = new AmendmentPatch();
    amendmentPatch.setOp(AmendmentPatchOperation.REMOVE);
    amendmentPatch.setPath(path);
    amendmentPatch.setValue(new AmendmentBooleanExtension(DrugsMarkerExtension.URL, "false"));
    return amendmentPatch;
}
````

Reply:
EXTENSION_PATH = "/extension/0"; defined in JsonPatches